python/ray/data/iterator.py

```diff
@@ -102,6 +102,7 @@ def _to_block_iterator(
         """
         raise NotImplementedError
 
+    @PublicAPI(stability="beta")
```

Comment:
maybe separate these annotations in another PR? it is not clear if these changes are related to the auto gen attempt